src/db/queries.rs

```diff
@@ -0,0 +1,196 @@
+use camino::{Utf8Path, Utf8PathBuf};
+use diesel::result::Error as DieselError;
+use diesel::SqliteConnection;
+
+use super::models::{AlbumRow, NewAlbumRow, NewSongRow, SongRow};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct AlbumId(i32);
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct SongId(i32);
+
+#[derive(Debug, Clone)]
+pub struct Album {
+    pub id: AlbumId,
+    pub directory: Utf8PathBuf,
+
+    pub title: Option<String>,
+    pub artist: Option<String>,
+    pub release_date: Option<String>,
+    pub original_art: Option<Utf8PathBuf>,
+    pub resized_art: Option<Utf8PathBuf>,
+}
+
+impl From<AlbumRow> for Album {
+    fn from(row: AlbumRow) -> Self {
+        Self {
+            id: AlbumId(row.id),
+            directory: row.directory.into(),
+            title: row.title,
+            artist: row.artist,
+            release_date: row.release_date,
+            original_art: row.original_art.map(Into::into),
+            resized_art: row.resized_art.map(Into::into),
+        }
+    }
+}
+
+impl Album {
+    pub fn display_title(&self) -> &str {
+        if let Some(title) = &self.title {
+            return title;
+        }
+
+        let directory_name = self.directory.components().last();
+
+        directory_name.map(|c| c.as_str()).unwrap_or_default()
```

Comment:
These methods should return an option, so the ui can decide how to handle not showing them.